R/tables.R

```diff
@@ -1,117 +1,274 @@
+  }
+}
+
+
+
+
+
+
+
+fix_classes <- function(table) {
+  logical_cols <- names(table)[names(table) %in% logical_vars]
+  date_cols <-  names(table)[names(table) %in% date_vars]
+  datetime_cols <- names(table)[names(table)  %in% datetime_vars]
+  if(length(logical_cols) != 0) table <- mutate_at(table, vars(logical_cols), funs_("as.logical"))
+  if(length(date_cols) != 0) table <- mutate_at(table, vars(date_cols), funs(quicktime))
+  if(length(datetime_cols) != 0) table <- mutate_at(table, vars(datetime_cols), funs(quicktime2))
+  return(table)
+}
+
+#' @export
+#' @rdname ed_table
+ed_table <- function(table, ...) {
+  ed_table_(table, .dots = lazyeval::lazy_dots(...))
+}
+
+#' @export
+#' @rdname ed_table
+ed_events <- function(...) {
+  ed_table_("events", .dots = lazyeval::lazy_dots(...))
+}
+
+#' @export
+#' @rdname ed_table
+ed_animals <- function(...) {
+  ed_table_("animals", .dots = lazyeval::lazy_dots(...))
+}
+
+#' @export
+#' @rdname ed_table
+ed_specimens <- function(...) {
+  ed_table_("specimens", .dots = lazyeval::lazy_dots(...))
+}
+
+#' @export
+#' @rdname ed_table
+ed_tests <- function(...) {
+  ed_table_("tests", .dots = lazyeval::lazy_dots(...))
+}
+
+#' @export
+#' @rdname ed_table
+ed_viruses <- function(...) {
+  ed_table_("viruses", .dots = lazyeval::lazy_dots(...))
+}
+
+#' @export
+#' @rdname ed_table
+ed_testspecimen <- function(...) {
+  ed_table_("test_specimen_ids", .dots = lazyeval::lazy_dots(...))
+}
+
+#' @export
+#' @importFrom DBI dbReadTable
+ed2_key_errors <- function() {
+  dbReadTable(eidith_db()$con, "p2_unique_id_errors")
+}
+
+#' @export
+ed2_events <- function(...){
+  ed_table_("events_2", .dots = lazyeval::lazy_dots(...))
+}
+
+#' @export
+ed2_animals <- function(...) {
+  ed_table_("animals_2", .dots = lazyeval::lazy_dots(...))
+}
+
+#' @export
+ed2_specimens <- function(...) {
+  ed_table_("specimens_2", .dots = lazyeval::lazy_dots(...))
+}
+
+#' @export
+ed2_animal_production <- function(...){
+  ed_table_("animal_production_2", .dots = lazyeval::lazy_dots(...))
+}
+
+#' @export
+ed2_crop_production <- function(...){
+  ed_table_("crop_production_2", .dots = lazyeval::lazy_dots(...))
+}
+
+#' @export
+ed2_zoo_sanctuary <- function(...){
+  ed_table_("zoo_sanctuary_2", .dots = lazyeval::lazy_dots(...))
+}
+
+#' @export
+ed2_wildlife_restaurant <- function(...){
+  ed_table_("wildlife_restaurant_2", .dots = lazyeval::lazy_dots(...))
+}
+
+#' @export
+ed2_natural_areas <- function(...){
+  ed_table_("natural_areas_2", .dots = lazyeval::lazy_dots(...))
+}
+
+#' @export
+ed2_market_value_chain <- function(...){
+  ed_table_("market_value_chain_2", .dots = lazyeval::lazy_dots(...))
+}
+
+#' @export
+ed2_human <- function(...){
+  ed_table_("human_2", .dots = lazyeval::lazy_dots(...))
+}
+
+#' @export
+ed2_tests <- function(...){
+  ed_table_("tests_2", .dots = lazyeval::lazy_dots(...))
+}
+
+#' @export
+ed2_test_serology <- function(...){

```

Comment:
I get the message that there are notes in this table despite the notes column being all `NA` values.

Reply:
Fixed to only print note fields if there's at least one non-NA note.